drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/ConditionPopupTest.java

```diff
@@ -89,343 +90,420 @@ public void setUp() throws Exception {
     }
 
     @Test
     public void testBindingDifferentFactAndField() {
-        column.setBinding( "$field" );
-        setUpPopup( model,
-                    pattern,
-                    column,
-                    false,
-                    false );
+        column.setBinding("$field");
+        setUpPopup(model,
+                   pattern,
+                   column,
+                   false,
+                   false);
 
-        popup.view = spy( new ConditionPopupView( popup ) ) ;
+        popup.view = spy(new ConditionPopupView(popup));
 
         popup.applyChanges();
-        verify( popup.view, never() ).warnAboutAlreadyUsedBinding();
+        verify(popup.view,
+               never()).warnAboutAlreadyUsedBinding();
     }
 
     @Test
     public void testDuplicateHeader() throws Exception {
-        column.setBinding( "$field" );
+        column.setBinding("$field");
 
-        BaseColumn baseColumn = mock( BaseColumn.class );
-        when( baseColumn.getHeader() ).thenReturn( COLUMN_HEADER );
-        CompositeColumn compositeColumn = mock( CompositeColumn.class );
-        when( compositeColumn.getChildColumns() ).thenReturn(Arrays.asList( baseColumn ) );
+        BaseColumn baseColumn = mock(BaseColumn.class);
+        when(baseColumn.getHeader()).thenReturn(COLUMN_HEADER);
+        CompositeColumn compositeColumn = mock(CompositeColumn.class);
+        when(compositeColumn.getChildColumns()).thenReturn(Arrays.asList(baseColumn));
 
-        model.getConditions().add( compositeColumn );
+        model.getConditions().add(compositeColumn);
 
-        setUpPopup( model,
-                    pattern,
-                    column,
-                    true,
-                    false );
+        setUpPopup(model,
+                   pattern,
+                   column,
+                   true,
+                   false);
 
-        popup.view = spy( new ConditionPopupView( popup ) ) ;
+        popup.view = spy(new ConditionPopupView(popup));
 
         popup.applyChanges();
-        verify( popup.view ).warnAboutAlreadyUsedColumnHeaderName();
+        verify(popup.view).warnAboutAlreadyUsedColumnHeaderName();
     }
 
     @Test
     public void testClearBindingWhenPredicate() throws Exception {
-        column.setBinding( "$field" );
-        column.setConstraintValueType( BaseSingleFieldConstraint.TYPE_PREDICATE );
+        column.setBinding("$field");
+        column.setConstraintValueType(BaseSingleFieldConstraint.TYPE_PREDICATE);
 
-        setUpPopup( model,
-                    pattern,
-                    column,
-                    true,
-                    false );
+        setUpPopup(model,
+                   pattern,
+                   column,
+                   true,
+                   false);
 
         popup.applyChanges();
-        assertNull( popup.getEditingCol().getBinding() );
+        assertNull(popup.getEditingCol().getBinding());
     }
 
+    @Test
```

Comment:
Is there a test for that checks the IllegalStateException? Would be good to check that situation.